rflink/parser.py

```diff
@@ -283,6 +311,36 @@ def encode_packet(packet: dict) -> str:
     ... })
     '10;newkaku;000001;01;on;'
     """
+    
+    """
+    TODO This template is not guaranteed to be the right one. For example, it ignores the RGBW value for Milight.
+
+    # 10;NewKaku;0cac142;3;ON;
+    PACKET_COMMAND = DELIM.join(['10', PROTOCOL, ADDRESS, BUTTON, COMMAND])
+    # 10;MiLightv1;F746;00;3c00;ON;
+    PACKET_COMMAND2 = DELIM.join(['10', PROTOCOL, ADDRESS, BUTTON, VALUE, COMMAND])
+    # 10;MERTIK;64;UP;
+    PACKET_COMMAND3 = DELIM.join(['10', PROTOCOL, ADDRESS, COMMAND])
+    # 10;DELTRONIC;001c33;
+    PACKET_COMMAND4 = DELIM.join(['10', PROTOCOL, ADDRESS])
+
+
+    Will use this one:
+    SWITCH_COMMAND_TEMPLATE_SWITCH_COMMAND = '{node};{protocol};{id};{switch};{command};'
+    SWITCH_COMMAND_TEMPLATE_SWITCH_VALUE_COMMAND = '{node};{protocol};{id};{switch};{value};{command};'
+    SWITCH_COMMAND_TEMPLATE_COMMAND = '{node};{protocol};{id};{command};'
+    SWITCH_COMMAND_TEMPLATE_MINIMAL = '{node};{protocol};{id};'
+    
+    """
+    if 'value' in packet:
```

Comment:
Instead of relying on the presence of a field we might want to make the 'type' of packet and explicit field and decide the template used based on that? 
```
COMMAND_TEMPLATES = {
'switch': '{node};{protocol};{id};{switch};{command};',
....
}

....

def encode_packet(packet: dict) -> str:
    command_template = COMMAND_TEMPLATES[packet['type']]

    return command_template.format........
```